### PR TITLE
Updated gherkin to 3.0 - this requires updating most of the implement…

### PR DESCRIPTION
**Pull request by sponte**
…ation due to parser changes

### CODE REVIEW

**Comment by sponte** [comment]
This is not ready yet. You're welcome to pick it up as I might need to work on something else on Monday

**Comment by jimCresswell** [comment]
Happy to pick it up.

I've only skimmed it, looks like it might be dropping support for multiple features in a file, is that enforced by Gherkin 3?

**Comment by jimCresswell** [comment]
Very nice in general. I'd like to pull the parser out into a separate module soonish.

**Comment by sponte** [comment]
Yes it is enforced. There are other assumptions we've made that are not
valid. For instance there is no such thing as scenario comments or multiple
step attributes (tables or docStrings). But that's ok. It's still early :)

>

-- 
Kind regards,
Stanislaw Wozniak

**Comment by jimCresswell** [comment]
Those all sound like good changes but they were also things picked up by
Gherkin 2 as valid (maybe there was more validation inside CucumberJS), we
should add a note to the readme about where our definition of valid comes
from :)

**Comment by jimCresswell** [comment]
I noticed the branch is also for general udpates to dependencies, given the work required for Gherkin 3 do you mind if we stick with just doing Cucumber in this branch?

**Comment by jimCresswell** [comment]
I'll do some work on this branch now...

**Comment by sponte** [comment]
That's fine. It was supposed to be update dependencies task as I've noticed Amber with David tool. It turned slightly into the moloch it is now. 

Kind regards,
Stanislaw Wozniak

* Sent from my iPhone

File: features-support/step_definitions/parsing.js [hunk 1]
```diff
@@ -29,7 +32,12 @@ module.exports = function() {
 
   function compareFeatureValues(key) {
     return function compare(table) {
-      var featureValues = features[0][key];
+      var featureValues = features[key].map(function(a) {
+        if(key == "comments")
```
Comment (jimCresswell) [inline features-support/step_definitions/parsing.js]: Rule of thumb in the JS community, either single line ifs with no brackets or use brackets on multilines :smile:

**Comment by jimCresswell** [comment]
So this is arguably ready to merge... but let's go over the changes, and make sure it all makes sense.

**Comment by sponte** [comment]
One of the feature files does not compile: https://oss-specs-specs-eu-pr-120.herokuapp.com/specs/features-support/test-data/example-feature-file.feature. And there tags and comments have trailing spaces (and probably leading)

**Comment by jimCresswell** [comment]
It's using master by default and on that branch that feature file is indeed
invalid because it contains 2 features... (you fixed it in this pull
request).

I'll take a look at the whitespace tomorrow.

**Comment by jimCresswell** [comment]
How's that?

Regarding the error messages, I'd like to leave them in their current format while we are in the rapid development phase.

**Pull request merged**